src/com/amazon/ionelement/api/IonElement.kt

```diff
@@ -425,6 +425,9 @@ public interface StructElement : ContainerElement {
     /** This struct's unordered collection of fields. */
     public val fields: Collection<StructField>
 
+    /** A mutable shallow copy of this struct's fields */
+    public val mutableFields: MutableStructFields
```

Comment:
I agree a function makes more sense since the property always returns a new object anyway.

Reply:
Another question is whether the MutableStructFields object can or should carry the annotations and metas attached to the original struct.

Reply:
If it's `MutableStructFields`, then I would say that it does not need the annotations or the metas. If we were going for a `MutableStructElement`, then we probably should include the annotations and metas.

I'm not opposed to going for a `MutableStructElement`, but I think it's a much more drastic change since we would need to define how the mutable elements fit into the inheritance hierarchy (does it implement `MutableIonElement`, and how do `IonElement` and `MutableIonElement` relate?) and we would probably end up implementing mutable versions for all of the Ion types.